HCollections/HListT.fs

```diff
@@ -71,6 +71,45 @@ module HListT =
                         xs |> Teq.castFrom teq
                 }
 
+    let rec private toHList'<'ts, 'elem, 'k>
+        (input : HListT<'ts, 'elem>)
```

Comment:
I'm now wondering if a `HListT` should be implemented as an `'elem list` and a `'ts HList` internally. We'd get these for free then. We already have a `TypeList` internally; why should we not use an `HList` internally?

Reply:
Can make the change

Reply:
Yes, might be worth trying this out to see how it looks. You'll lose some safety in the implementation (as now we are using the API only to guarantee that the `'elem list` and the `'ts HList` are the same length, but you'll gain performance wins for when you call `toHList` and `toList`

Reply:
N.B if you do want to try changing the implementation, I would suggest getting this merged first and then doing it as another PR.

Reply:
Sounds good to me, here's the updated HListT: https://github.com/JackMatusiewicz/HCollections/tree/HListTStoresHList

Once this has been merged I can fix the new branch up.